spec/app/domain/authentication/authn-azure/validate_application_identity_spec.rb

```diff
@@ -0,0 +1,428 @@
+
+  let(:mismatched_subscription_id_annotation) { double("MismatchedSubscriptionIdAnnotation") }
+  let(:mismatched_resource_group_annotation) { double("MismatchedResourceGroupAnnotation") }
+  let(:mismatched_system_assigned_identity_annotation) { double("MismatchedSystemAssignedAnnotation") }
+  let(:mismatched_user_assigned_identity_annotation) { double("MismatchedUserAssignedAnnotation") }
+
+  let(:validate_azure_annotations) { double("ValidateAzureAnnotations") }
+
+  def mock_annotation_builder(host_annotation_type, host_annotation_key, host_annotation_value)
+    allow(host_annotation_type).to receive(:values)
+                                     .and_return(host_annotation_type)
+    allow(host_annotation_type).to receive(:[])
+                                     .with(:name)
+                                     .and_return(host_annotation_key)
+    allow(host_annotation_type).to receive(:[])
+                                     .with(:value)
+                                     .and_return(host_annotation_value)
+  end
+
+  before(:each) do
+    allow(host).to receive(:annotations)
+                     .and_return([subscription_id_annotation, resource_group_annotation])
+
+    allow(validate_azure_annotations).to receive(:call)
+                                           .and_return(true)
+
+    allow(resource_class).to receive(:[])
+                               .with("#{account}:host:#{hostname}")
+                               .and_return(host)
+
+    mock_annotation_builder(subscription_id_annotation,"authn-azure/subscription-id", "some-subscription-id-value")
+    mock_annotation_builder(resource_group_annotation,"authn-azure/resource-group", "some-resource-group-value")
+    mock_annotation_builder(user_assigned_identity_annotation,"authn-azure/user-assigned-identity", "some-user-assigned-identity-value")
+    mock_annotation_builder(system_assigned_identity_annotation,"authn-azure/system-assigned-identity", oid_token_field)
+    mock_annotation_builder(mismatched_subscription_id_annotation, "authn-azure/subscription-id", "mismatched-subscription-id")
+    mock_annotation_builder(mismatched_resource_group_annotation, "authn-azure/resource-group", "mismatched-resource-group")
+    mock_annotation_builder(mismatched_user_assigned_identity_annotation, "authn-azure/user-assigned-identity", "mismatched-user-assigned-identity")
+    mock_annotation_builder(mismatched_system_assigned_identity_annotation, "authn-azure/system-assigned-identity", "mismatched-system-assigned-identity")
+  end
+
+
+  #  ____  _   _  ____    ____  ____  ___  ____  ___
+  # (_  _)( )_( )( ___)  (_  _)( ___)/ __)(_  _)/ __)
+  #   )(   ) _ (  )__)     )(   )__) \__ \  )(  \__ \
+  #  (__) (_) (_)(____)   (__) (____)(___/ (__) (___/
+
+  context "A valid xms_mirid claim in the Azure token" do
+    context "with a valid application identity" do
+      context "and a non-existent Role in Conjur" do
```

Comment:
we have the vanilla tests first. please align your UTs with this.

Reply:
swapped "and a non-existent Role in Conjur" and "and an Azure token with matching data"